ckan/authz.py

```diff
@@ -119,6 +121,8 @@ def _get_user(username):
     except TypeError:
         # c is not available
         pass
+    except DetachedInstanceError:
```

Comment:
Yes, that looks promising. Passing the username in the context is how most action functions expect to find it, no need to fetch the user yourself.